### PR TITLE
Campaign/applytheme sidebar header text color

### PR DESCRIPTION
**Pull request by deanwhillier**
#### Summary
ApplyTheme Campaign: This PR completes the merge of the theme.sidebarHeaderTextColor commits from the campaign branch.

Dev approval for the changes was completed on the original PR(s), so we are just looking for rubber stamp approvals by dev's to satisfy merge requirements.

As QA was skipped on PR merges into campaign branches, this PR will need proper QA before merging. See below for details on what to QA for this PR.

#### Ticket Link

* https://github.com/mattermost/mattermost-server/issues/15871
* https://github.com/mattermost/mattermost-server/issues/15872
* https://github.com/mattermost/mattermost-server/issues/15873
* https://github.com/mattermost/mattermost-server/issues/15874
* https://github.com/mattermost/mattermost-server/issues/15875
* https://github.com/mattermost/mattermost-server/issues/15876

#### Related Pull Requests

See tickets for related pull requests

### CODE REVIEW

File: sass/layout/_navigation.scss [hunk 1]
```diff
@@ -137,6 +138,10 @@
             padding: 0;
             display: flex;
 
+            .app__body & {
+                color: var(--sidebar-header-text-color)
+            }
```
Comment (deanwhillier) [inline sass/layout/_navigation.scss:143]: @jgilliam17, this affects the text colour of the header bar in mobile view

File: sass/responsive/_mobile.scss [hunk 2]
```diff
@@ -1370,7 +1370,7 @@
     .search-bar__container {
         @include flex(0 0 50px);
         background: $primary-color;
-        color: $white;
+        color: var(--sidebar-header-text-color);
```
Comment (deanwhillier) [inline sass/responsive/_mobile.scss:1373]: @jgilliam17, this is supposed to set the text colour for the search bar header in mobile, but unfortunately the input field doesn't inherit it and so is not being styled properly. We have a separate ticket to handle that.

**Review by devinbinnie — APPROVED** [review]

**Review by nevyangelova — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks for the test steps @deanwhillier 
Tested, looks good to merge.
- Verified background color of navbar buttons, header bar in mobile view - as expected. Text color is search bar in mobile view is not fixed here, will be handled in a different ticket.

**Pull request merged**